src/coreclr/scripts/genEventing.py

```diff
@@ -27,28 +39,100 @@
-palDataTypeMapping ={
-        #constructed types
-        "win:null"          :" ",
-        "win:Int64"         :"const __int64",
-        "win:ULong"         :"const ULONG",
-        "win:count"         :"*",
-        "win:Struct"        :"const void",
-        #actual spec
-        "win:GUID"          :"const GUID",
-        "win:AnsiString"    :"LPCSTR",
-        "win:UnicodeString" :"PCWSTR",
-        "win:Double"        :"const double",
-        "win:Int32"         :"const signed int",
-        "win:Boolean"       :"const BOOL",
-        "win:UInt64"        :"const unsigned __int64",
-        "win:UInt32"        :"const unsigned int",
-        "win:UInt16"        :"const unsigned short",
-        "win:UInt8"         :"const unsigned char",
-        "win:Pointer"       :"const void*",
-        "win:Binary"        :"const BYTE"
-        }
+lindent = "    "
+
+coreCLRPalDataTypeMapping={
+    #constructed types
+    "win:null"          :" ",
+    "win:Int64"         :"const __int64",
+    "win:ULong"         :"const ULONG",
+    "win:count"         :"*",
+    "win:Struct"        :"const void",
+    #actual spec
+    "win:GUID"          :"const GUID",
+    "win:AnsiString"    :"LPCSTR",
+    "win:UnicodeString" :"PCWSTR",
+    "win:Double"        :"const double",
+    "win:Int32"         :"const signed int",
+    "win:Boolean"       :"const BOOL",
+    "win:UInt64"        :"const unsigned __int64",
+    "win:UInt32"        :"const unsigned int",
+    "win:UInt16"        :"const unsigned short",
+    "win:UInt8"         :"const unsigned char",
+    "win:Pointer"       :"const void*",
+    "win:Binary"        :"const BYTE",
+}
+
+coreCLREventPipeDataTypeMapping={
+    "BOOL"              : "BOOL",
+    "LPCGUID"           : "LPCGUID",
+    "UCHAR"             : "UCHAR",
+    "ULONG"             : "ULONG",
+    "ULONGLONG"         : "ULONGLONG",
+    "WCHAR"             : "WCHAR",
+    "BYTE"              : "BYTE",
+}
+
+monoPalDataTypeMapping={
+    #constructed types
+    "win:null"          :" ",
+    "win:Int64"         :"const int64_t",
+    "win:ULong"         :"const uint32_t",
+    "win:count"         :"*",
+    "win:Struct"        :"const void",
+    #actual spec
+    "win:GUID"          :"const uint8_t",
+    "win:AnsiString"    :"const char*",
+    "win:UnicodeString" :"const ep_char8_t*",
+    "win:Double"        :"const double",
+    "win:Int32"         :"const int32_t",
+    "win:Boolean"       :"const bool",
+    "win:UInt64"        :"const uint64_t",
+    "win:UInt32"        :"const uint32_t",
+    "win:UInt16"        :"const uint16_t",
+    "win:UInt8"         :"const uint8_t",
+    "win:Pointer"       :"const void*",
+    "win:Binary"        :"const uint8_t",
+}
+
+monoEventPipeDataTypeMapping={
+    "BOOL"              : "bool",
+    "LPCGUID"           : "const uint8_t*",
+    "UCHAR"             : "uint8_t",
+    "ULONG"             : "uint32_t",
+    "ULONGLONG"         : "uint64_t",
+    "WCHAR"             : "wchar_t",
```

Comment:
not `ep_char16_t` ?

Reply:
This is actually used for some strings in generated scripts that are using WCHAR in coreclr. Since this string can That mapping corresponds to that and we have code that does conversion to UTF8 on Mono side (only used for provider name).